diff -urN squirrelmail-1.4.4.orig/plugins/squirrelspell/sqspell_config.php squirrelmail-1.4.4/plugins/squirrelspell/sqspell_config.php
--- squirrelmail-1.4.4.orig/plugins/squirrelspell/sqspell_config.php	2004-12-27 16:03:58.000000000 +0100
+++ squirrelmail-1.4.4/plugins/squirrelspell/sqspell_config.php	2005-05-30 18:35:01.000000000 +0200
@@ -24,6 +24,7 @@
  *                     'Spanish' => 'ispell -d spanish -a' );
  */
 $SQSPELL_APP = array('English' => 'ispell -a',
+			'Polish' => 'ispell -d polish -a',
 			'Spanish' => 'ispell -d spanish -a');
 $SQSPELL_APP_DEFAULT = 'English';
 $SQSPELL_WORDS_FILE = 
