diff -urN squirrelmail-1.4.5.orig/class/mime/Rfc822Header.class.php squirrelmail-1.4.5/class/mime/Rfc822Header.class.php
--- squirrelmail-1.4.5.orig/class/mime/Rfc822Header.class.php	2005-02-07 01:33:29.000000000 +0100
+++ squirrelmail-1.4.5/class/mime/Rfc822Header.class.php	2005-09-27 13:44:21.000000000 +0200
@@ -505,8 +505,9 @@
      * functions/imap_messages. I'm not sure if it's ok here to call
      * that function?
      */
-    function parsePriority($value) {
-        $value = strtolower(array_shift(split('/\w/',trim($value))));
+    function parsePriority($sValue) {
+        $sValue = split('/\w/',trim($value));
+        $value = strtolower(array_shift($sValue));
         if ( is_numeric($value) ) {
             return $value;
         }
diff -urN squirrelmail-1.4.5.orig/functions/imap_messages.php squirrelmail-1.4.5/functions/imap_messages.php
--- squirrelmail-1.4.5.orig/functions/imap_messages.php	2005-04-16 19:45:38.000000000 +0200
+++ squirrelmail-1.4.5/functions/imap_messages.php	2005-09-27 12:37:43.000000000 +0200
@@ -476,8 +476,9 @@
  * NOTE: this is actually a duplicate from the function in
  * class/mime/Rfc822Header.php.
  */
-function parsePriority($value) {
-    $value = strtolower(array_shift(split('/\w/',trim($value))));
+function parsePriority($sValue) {
+	$aValue=split('/w/',trim($sValue));
+	$value = strtolower(array_shift($aValue));
     if ( is_numeric($value) ) {
         return $value;
     }
