--- squirrelmail-1.2.8/plugins/fortune/setup.php.orig	Mon Mar  5 16:21:41 2001
+++ squirrelmail-1.2.8/plugins/fortune/setup.php	Wed Sep 18 16:50:05 2002
@@ -12,7 +12,6 @@
   $squirrelmail_plugin_hooks["options_display_inside"]['fortune'] = 'fortune_options';
   $squirrelmail_plugin_hooks['options_display_save']['fortune'] = 'fortune_save';
   $squirrelmail_plugin_hooks['loading_prefs']['fortune'] = 'fortune_load';
-  
 }
 
 function fortune() {
@@ -20,7 +19,7 @@
   if( !$fortune_visible ) {
     return;
   }
-  $fortune_location = "/usr/games/fortune";
+  $fortune_location = "/usr/bin/fortune";
   global $color;
   $exist = file_exists("$fortune_location");
   echo "<center><table cellpadding=0 cellspacing=0 border=0 bgcolor=$color[10]><tr><td><table width=100% cellpadding=2 cellspacing=1 border=0 bgcolor=\"$color[5]\"><tr><td align=center>";
@@ -41,10 +40,8 @@
   global $fortune_visible;
 
   $fortune_visible = getPref($data_dir, $username, 'fortune_visible');
-  
 }
 
-
 function fortune_options() {
   global $fortune_visible;
   echo "<tr><td align=right nowrap>Fortunes:</td>\n";
@@ -54,18 +51,16 @@
   echo "> Show fortunes at top of mailbox</td></tr>\n";
 }
 
-function fortune_save() 
-{
-  global $username,$data_dir;
-  global $fortune_fortune_visible;
-
-  if (isset($fortune_fortune_visible)) 
-  {
-    setPref($data_dir, $username, 'fortune_visible', '1');
-  } 
-  else 
-  {
-    setPref($data_dir, $username, 'fortune_visible', '');
-  }
+function fortune_save() {
+    global $username,$data_dir;
+    if ( (float)substr(PHP_VERSION,0,3) < 4.1 ) {
+        global $_POST;
+    }
+ 
+    if(isset($_POST['fortune_fortune_visible'])) {
+        setPref($data_dir, $username, 'fortune_visible', '1');
+    } else {
+        setPref($data_dir, $username, 'fortune_visible', '');
+    }
 }
 ?>
