diff -ru squirrelmail-1.4.2/plugins/change_pass/options.php squirrelmail-1.4.2-AX/plugins/change_pass/options.php
--- squirrelmail-1.4.2/plugins/change_pass/options.php	2003-04-25 16:41:20.000000000 +0200
+++ squirrelmail-1.4.2-AX/plugins/change_pass/options.php	2004-02-27 20:00:38.000000000 +0100
@@ -50,7 +50,7 @@
     }
 ?>
                 <tr><td align="center" colspan="2">
-                    <input type="submit" value="Submit" name="plugin_change_pass">
+                    <input type="submit" value="<? echo _("Submit"); ?>" name="plugin_change_pass">
                 </td></tr>
             </table>
             </form>
@@ -60,7 +60,7 @@
 <?php
 function change_pass_closeport($pop_socket, &$messages, $debug) {
     if ($debug) {
-        array_push($messages, "Closing Connection");
+        array_push($messages, _("Closing Connection"));
     }
     fputs($pop_socket, "quit\r\n");
     fclose($pop_socket);
@@ -117,7 +117,7 @@
     $messages = array();
 
     if ($debug) {
-        $messages[] = "Connecting to Password Server";
+        $messages[] = _("Connecting to Password Server");
     }
     $pop_socket = fsockopen($imapServerAddress, 106, $errno, $errstr);
     if (!$pop_socket) {
@@ -152,7 +152,7 @@
         return $messages;
 
     //if ($debug) {
-        $messages[] = 'Password changed successfully.';
+        $messages[] = _("Password changed successfully.");
     //}
     // Write new cookies for the password
     $onetimepad = OneTimePadCreate(strlen($cp_newpass));
diff -ru squirrelmail-1.4.2/plugins/change_pass/setup.php squirrelmail-1.4.2-AX/plugins/change_pass/setup.php
--- squirrelmail-1.4.2/plugins/change_pass/setup.php	2003-04-04 00:49:59.000000000 +0200
+++ squirrelmail-1.4.2-AX/plugins/change_pass/setup.php	2004-02-27 20:00:56.000000000 +0100
@@ -20,7 +20,7 @@
 
 function change_pass_save_pref() {
     if(sqgetGlobalVar('plugin_change_pass', $plugin_change_pass)) {
-        echo "<p align=center>Password changed successfully.</p>\n";
+        echo "<p align=center>" . _("Password changed successfully.") . "</p>\n";
     }
 }
 ?>
