diff -urN squirrelmail-1.2.0-rc2.org/functions/i18n.php squirrelmail-1.2.0-rc2/functions/i18n.php
--- squirrelmail-1.2.0-rc2.org/functions/i18n.php	Thu Oct  4 14:15:24 2001
+++ squirrelmail-1.2.0-rc2/functions/i18n.php	Thu Oct  4 14:16:53 2001
@@ -829,6 +829,11 @@
       if (isset($sm_language) && $use_gettext &&
           $squirrelmail_language != '' &&
  	  isset($languages[$sm_notAlias]['CHARSET'])) {
+	 if (isset($sm_notAlias) && strlen($sm_notAlias) == 2
+	 	&& !is_file("/usr/share/i18n/locales/" . $sm_notAlias)) {
+	  $new_sm_notAlias = $sm_notAlias . "_" . strtoupper($sm_notAlias);
+	  if (is_file("/usr/share/i18n/locales/" . $new_sm_notAlias))
+	   $sm_notAlias = $new_sm_notAlias; }
          if ((ini_get('safe_mode') == FALSE) && (getenv('LC_ALL') != $sm_language)) {
            putenv('LC_ALL=' . $sm_notAlias);
          }
