--- squirrelmail-1.2.8/plugins/auto_cc/setup.php.orig	Sat Apr 21 03:21:31 2001
+++ squirrelmail-1.2.8/plugins/auto_cc/setup.php	Fri Sep 20 10:02:06 2002
@@ -70,15 +70,16 @@
    global $auto_cc_cc_addr;
    global $auto_cc_bcc_addr;
    global $auto_cc_cc_addri, $auto_cc_bcc_addri;
+   global $_POST;
 
-   if(isset($auto_cc_cc_addri)) {
-     setPref($data_dir, $username, "auto_cc_cc_addr", $auto_cc_cc_addri);
+   if(isset($_POST['auto_cc_cc_addri'])) {
+     setPref($data_dir, $username, "auto_cc_cc_addr", $_POST['auto_cc_cc_addri']);
    } else {
      setPref($data_dir, $username, "auto_cc_cc_addr", "");
    }
 
-   if(isset($auto_cc_bcc_addri)) {
-     setPref($data_dir, $username, "auto_cc_bcc_addr", $auto_cc_bcc_addri);
+   if(isset($_POST['auto_cc_bcc_addri'])) {
+     setPref($data_dir, $username, "auto_cc_bcc_addr", $_POST['auto_cc_bcc_addri']);
    } else {
      setPref($data_dir, $username, "auto_cc_bcc_addr", "");
    }
