--- squirrelmail-1.2.8/plugins/mail_fwd/setup.php.orig	Mon Mar  5 16:21:41 2001
+++ squirrelmail-1.2.8/plugins/mail_fwd/setup.php	Fri Sep 20 22:10:13 2002
@@ -37,14 +37,15 @@
 
 function mail_fwd_save_pref() {
   global $username,$data_dir;
-  global $mfwd_user;
+  global $mfwd_user;
+  global $_POST;
 
-  if (isset($mfwd_user)) {
-    setPref($data_dir,$username,"mailfwd_user",$mfwd_user);
+  if (isset($_POST['mfwd_user'])) {
+    setPref($data_dir,$username,"mailfwd_user", $_POST['mfwd_user']);
   } else {
     setPref($data_dir,$username,"mailfwd_user","");
   }
-  exec("/usr/sbin/wfwd ".$username." ".$mfwd_user);
+  exec("/usr/sbin/wfwd ".$username." ".$_POST['mfwd_user']);
 }
 
 ?>
--- squirrelmail-1.2.8/plugins/mail_fwd/fwdfile/wfwd.c.orig	Mon Mar  5 16:21:41 2001
+++ squirrelmail-1.2.8/plugins/mail_fwd/fwdfile/wfwd.c	Fri Sep 20 17:48:18 2002
@@ -49,7 +49,7 @@
      }
      setuid (pw->pw_uid);
      setgid (pw->pw_gid);
-     sprintf(line,"/home/%s/.forward",puid);
+     sprintf(line,"%s/.forward", pw->pw_dir);
      if ((fd = (FILE *)fopen(line,"w"))==NULL) {
         printf("Cannot open %s\n",line);
         exit(1);
