diff -uNr squirrelmail-1.2.7/index.php squirrelmail-1.2.7.new/index.php
--- squirrelmail-1.2.7/index.php	Tue Feb 19 16:05:03 2002
+++ squirrelmail-1.2.7.new/index.php	Wed Aug 28 10:27:14 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * index.php -- Displays the main frameset
  *
diff -uNr squirrelmail-1.2.7/src/addrbook_popup.php squirrelmail-1.2.7.new/src/addrbook_popup.php
--- squirrelmail-1.2.7/src/addrbook_popup.php	Sun Dec 23 08:42:40 2001
+++ squirrelmail-1.2.7.new/src/addrbook_popup.php	Wed Aug 28 10:34:00 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * addrbook_popup.php
  *
diff -uNr squirrelmail-1.2.7/src/addrbook_search.php squirrelmail-1.2.7.new/src/addrbook_search.php
--- squirrelmail-1.2.7/src/addrbook_search.php	Sat Mar  2 21:31:04 2002
+++ squirrelmail-1.2.7.new/src/addrbook_search.php	Wed Aug 28 10:33:57 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * addrbook_search.php
  *
diff -uNr squirrelmail-1.2.7/src/addrbook_search_html.php squirrelmail-1.2.7.new/src/addrbook_search_html.php
--- squirrelmail-1.2.7/src/addrbook_search_html.php	Wed May 15 17:17:22 2002
+++ squirrelmail-1.2.7.new/src/addrbook_search_html.php	Wed Aug 28 10:33:53 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * addrbook_search_html.php
  *
diff -uNr squirrelmail-1.2.7/src/addressbook.php squirrelmail-1.2.7.new/src/addressbook.php
--- squirrelmail-1.2.7/src/addressbook.php	Wed Jun 12 16:38:16 2002
+++ squirrelmail-1.2.7.new/src/addressbook.php	Wed Aug 28 10:33:49 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * addressbook.php
  *
diff -uNr squirrelmail-1.2.7/src/compose.php squirrelmail-1.2.7.new/src/compose.php
--- squirrelmail-1.2.7/src/compose.php	Fri Jun 21 03:37:58 2002
+++ squirrelmail-1.2.7.new/src/compose.php	Wed Aug 28 10:29:20 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * compose.php
  *
diff -uNr squirrelmail-1.2.7/src/delete_message.php squirrelmail-1.2.7.new/src/delete_message.php
--- squirrelmail-1.2.7/src/delete_message.php	Wed Feb 20 10:39:50 2002
+++ squirrelmail-1.2.7.new/src/delete_message.php	Wed Aug 28 10:29:02 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * delete_message.php
  *
diff -uNr squirrelmail-1.2.7/src/download.php squirrelmail-1.2.7.new/src/download.php
--- squirrelmail-1.2.7/src/download.php	Fri Mar  8 22:23:26 2002
+++ squirrelmail-1.2.7.new/src/download.php	Wed Aug 28 10:30:27 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * download.php
  *
diff -uNr squirrelmail-1.2.7/src/draft_actions.php squirrelmail-1.2.7.new/src/draft_actions.php
--- squirrelmail-1.2.7/src/draft_actions.php	Thu May 16 10:20:01 2002
+++ squirrelmail-1.2.7.new/src/draft_actions.php	Wed Aug 28 10:33:42 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * draft_actions.php
  *
diff -uNr squirrelmail-1.2.7/src/empty_trash.php squirrelmail-1.2.7.new/src/empty_trash.php
--- squirrelmail-1.2.7/src/empty_trash.php	Wed Jan 23 19:12:37 2002
+++ squirrelmail-1.2.7.new/src/empty_trash.php	Wed Aug 28 10:33:35 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * empty_trash.php
  *
diff -uNr squirrelmail-1.2.7/src/folders.php squirrelmail-1.2.7.new/src/folders.php
--- squirrelmail-1.2.7/src/folders.php	Tue Jun 11 04:25:55 2002
+++ squirrelmail-1.2.7.new/src/folders.php	Wed Aug 28 10:28:27 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * folders.php
  *
diff -uNr squirrelmail-1.2.7/src/folders_create.php squirrelmail-1.2.7.new/src/folders_create.php
--- squirrelmail-1.2.7/src/folders_create.php	Sun Mar  3 11:56:41 2002
+++ squirrelmail-1.2.7.new/src/folders_create.php	Wed Aug 28 10:33:30 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * folders_create.php
  *
diff -uNr squirrelmail-1.2.7/src/folders_delete.php squirrelmail-1.2.7.new/src/folders_delete.php
--- squirrelmail-1.2.7/src/folders_delete.php	Sun Mar  3 11:56:41 2002
+++ squirrelmail-1.2.7.new/src/folders_delete.php	Wed Aug 28 10:33:26 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * folders_delete.php
  *
diff -uNr squirrelmail-1.2.7/src/folders_rename_do.php squirrelmail-1.2.7.new/src/folders_rename_do.php
--- squirrelmail-1.2.7/src/folders_rename_do.php	Sat Feb  2 18:44:21 2002
+++ squirrelmail-1.2.7.new/src/folders_rename_do.php	Wed Aug 28 10:33:23 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * folders_rename_do.php
  *
diff -uNr squirrelmail-1.2.7/src/folders_rename_getname.php squirrelmail-1.2.7.new/src/folders_rename_getname.php
--- squirrelmail-1.2.7/src/folders_rename_getname.php	Sun Mar  3 11:56:41 2002
+++ squirrelmail-1.2.7.new/src/folders_rename_getname.php	Wed Aug 28 10:33:19 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * folders_rename_getname.php
  *
diff -uNr squirrelmail-1.2.7/src/folders_subscribe.php squirrelmail-1.2.7.new/src/folders_subscribe.php
--- squirrelmail-1.2.7/src/folders_subscribe.php	Mon Jan 28 19:33:33 2002
+++ squirrelmail-1.2.7.new/src/folders_subscribe.php	Wed Aug 28 10:33:15 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * folders_subscribe.php
  *
diff -uNr squirrelmail-1.2.7/src/help.php squirrelmail-1.2.7.new/src/help.php
--- squirrelmail-1.2.7/src/help.php	Thu May 16 10:28:06 2002
+++ squirrelmail-1.2.7.new/src/help.php	Wed Aug 28 10:30:17 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * help.php
  *
diff -uNr squirrelmail-1.2.7/src/image.php squirrelmail-1.2.7.new/src/image.php
--- squirrelmail-1.2.7/src/image.php	Tue Feb 19 13:31:12 2002
+++ squirrelmail-1.2.7.new/src/image.php	Wed Aug 28 10:33:10 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * image.php
  *
diff -uNr squirrelmail-1.2.7/src/index.php squirrelmail-1.2.7.new/src/index.php
--- squirrelmail-1.2.7/src/index.php	Sun Dec 23 08:42:40 2001
+++ squirrelmail-1.2.7.new/src/index.php	Wed Aug 28 10:33:06 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * index.php
  *
diff -uNr squirrelmail-1.2.7/src/left_main.php squirrelmail-1.2.7.new/src/left_main.php
--- squirrelmail-1.2.7/src/left_main.php	Wed May 29 00:53:57 2002
+++ squirrelmail-1.2.7.new/src/left_main.php	Wed Aug 28 10:29:45 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * left_main.php
  *
diff -uNr squirrelmail-1.2.7/src/load_prefs.php squirrelmail-1.2.7.new/src/load_prefs.php
--- squirrelmail-1.2.7/src/load_prefs.php	Wed Apr 17 16:41:46 2002
+++ squirrelmail-1.2.7.new/src/load_prefs.php	Wed Aug 28 10:32:59 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * load_prefs.php
  *
diff -uNr squirrelmail-1.2.7/src/login.php squirrelmail-1.2.7.new/src/login.php
--- squirrelmail-1.2.7/src/login.php	Thu May 16 10:18:53 2002
+++ squirrelmail-1.2.7.new/src/login.php	Wed Aug 28 10:28:01 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * login.php -- simple login screen
  *
diff -uNr squirrelmail-1.2.7/src/move_messages.php squirrelmail-1.2.7.new/src/move_messages.php
--- squirrelmail-1.2.7/src/move_messages.php	Wed May 29 04:28:54 2002
+++ squirrelmail-1.2.7.new/src/move_messages.php	Wed Aug 28 10:32:54 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * move_messages.php
  *
diff -uNr squirrelmail-1.2.7/src/options.php squirrelmail-1.2.7.new/src/options.php
--- squirrelmail-1.2.7/src/options.php	Sat Mar  2 21:31:04 2002
+++ squirrelmail-1.2.7.new/src/options.php	Wed Aug 28 10:32:49 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * options.php
  *
diff -uNr squirrelmail-1.2.7/src/options_display.php squirrelmail-1.2.7.new/src/options_display.php
--- squirrelmail-1.2.7/src/options_display.php	Wed Apr 17 20:52:22 2002
+++ squirrelmail-1.2.7.new/src/options_display.php	Wed Aug 28 10:32:45 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * options_display.php
  *
diff -uNr squirrelmail-1.2.7/src/options_folder.php squirrelmail-1.2.7.new/src/options_folder.php
--- squirrelmail-1.2.7/src/options_folder.php	Fri Jan  4 16:27:07 2002
+++ squirrelmail-1.2.7.new/src/options_folder.php	Wed Aug 28 10:32:41 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * options_folder.php
  *
diff -uNr squirrelmail-1.2.7/src/options_highlight.php squirrelmail-1.2.7.new/src/options_highlight.php
--- squirrelmail-1.2.7/src/options_highlight.php	Wed Apr 17 17:36:21 2002
+++ squirrelmail-1.2.7.new/src/options_highlight.php	Wed Aug 28 10:32:38 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * options_highlight.php
  *
diff -uNr squirrelmail-1.2.7/src/options_identities.php squirrelmail-1.2.7.new/src/options_identities.php
--- squirrelmail-1.2.7/src/options_identities.php	Thu Feb 14 07:12:51 2002
+++ squirrelmail-1.2.7.new/src/options_identities.php	Wed Aug 28 10:32:35 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * options_identities.php
  *
diff -uNr squirrelmail-1.2.7/src/options_order.php squirrelmail-1.2.7.new/src/options_order.php
--- squirrelmail-1.2.7/src/options_order.php	Tue Feb 19 13:31:12 2002
+++ squirrelmail-1.2.7.new/src/options_order.php	Wed Aug 28 10:32:30 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * options_order.php
  *
diff -uNr squirrelmail-1.2.7/src/options_personal.php squirrelmail-1.2.7.new/src/options_personal.php
--- squirrelmail-1.2.7/src/options_personal.php	Sun May  5 13:11:33 2002
+++ squirrelmail-1.2.7.new/src/options_personal.php	Wed Aug 28 10:32:25 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * options_personal.php
  *
diff -uNr squirrelmail-1.2.7/src/printer_friendly_bottom.php squirrelmail-1.2.7.new/src/printer_friendly_bottom.php
--- squirrelmail-1.2.7/src/printer_friendly_bottom.php	Fri Jun 21 04:00:00 2002
+++ squirrelmail-1.2.7.new/src/printer_friendly_bottom.php	Wed Aug 28 10:32:18 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * printer_friendly_bottom.php
  *
diff -uNr squirrelmail-1.2.7/src/printer_friendly_main.php squirrelmail-1.2.7.new/src/printer_friendly_main.php
--- squirrelmail-1.2.7/src/printer_friendly_main.php	Sun Apr  7 05:51:39 2002
+++ squirrelmail-1.2.7.new/src/printer_friendly_main.php	Wed Aug 28 10:32:14 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * printer_friendly_main.php
  *
diff -uNr squirrelmail-1.2.7/src/printer_friendly_top.php squirrelmail-1.2.7.new/src/printer_friendly_top.php
--- squirrelmail-1.2.7/src/printer_friendly_top.php	Tue Apr  2 12:05:33 2002
+++ squirrelmail-1.2.7.new/src/printer_friendly_top.php	Wed Aug 28 10:32:11 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * printer_friendly_top.php
  *
diff -uNr squirrelmail-1.2.7/src/read_body.php squirrelmail-1.2.7.new/src/read_body.php
--- squirrelmail-1.2.7/src/read_body.php	Mon Jun 10 20:16:32 2002
+++ squirrelmail-1.2.7.new/src/read_body.php	Wed Aug 28 10:30:34 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * read_body.php
  *
diff -uNr squirrelmail-1.2.7/src/redirect.php squirrelmail-1.2.7.new/src/redirect.php
--- squirrelmail-1.2.7/src/redirect.php	Sun Apr  7 00:44:48 2002
+++ squirrelmail-1.2.7.new/src/redirect.php	Wed Aug 28 10:32:06 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
 * redirect.php
 * Derived from webmail.php by Ralf Kraudelt <kraude@wiwi.uni-rostock.de>
diff -uNr squirrelmail-1.2.7/src/retrievalerror.php squirrelmail-1.2.7.new/src/retrievalerror.php
--- squirrelmail-1.2.7/src/retrievalerror.php	Fri Jun 21 20:42:17 2002
+++ squirrelmail-1.2.7.new/src/retrievalerror.php	Wed Aug 28 10:32:02 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * retrievalerror.php
  *
diff -uNr squirrelmail-1.2.7/src/right_main.php squirrelmail-1.2.7.new/src/right_main.php
--- squirrelmail-1.2.7/src/right_main.php	Fri May 24 17:35:44 2002
+++ squirrelmail-1.2.7.new/src/right_main.php	Wed Aug 28 10:28:07 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * right_main.php
  *
diff -uNr squirrelmail-1.2.7/src/search.php squirrelmail-1.2.7.new/src/search.php
--- squirrelmail-1.2.7/src/search.php	Tue Jun 11 04:25:55 2002
+++ squirrelmail-1.2.7.new/src/search.php	Wed Aug 28 10:28:15 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * search.php
  *
diff -uNr squirrelmail-1.2.7/src/signout.php squirrelmail-1.2.7.new/src/signout.php
--- squirrelmail-1.2.7/src/signout.php	Fri Apr 12 05:10:29 2002
+++ squirrelmail-1.2.7.new/src/signout.php	Wed Aug 28 10:31:52 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * signout.php -- cleans up session and logs the user out
  *
diff -uNr squirrelmail-1.2.7/src/validate.php squirrelmail-1.2.7.new/src/validate.php
--- squirrelmail-1.2.7/src/validate.php	Mon Apr  8 22:35:29 2002
+++ squirrelmail-1.2.7.new/src/validate.php	Wed Aug 28 10:31:48 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
 * validate.php
 *
diff -uNr squirrelmail-1.2.7/src/vcard.php squirrelmail-1.2.7.new/src/vcard.php
--- squirrelmail-1.2.7/src/vcard.php	Tue Feb 19 13:31:13 2002
+++ squirrelmail-1.2.7.new/src/vcard.php	Wed Aug 28 10:31:45 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * vcard.php
  *
diff -uNr squirrelmail-1.2.7/src/webmail.php squirrelmail-1.2.7.new/src/webmail.php
--- squirrelmail-1.2.7/src/webmail.php	Fri May 24 18:12:42 2002
+++ squirrelmail-1.2.7.new/src/webmail.php	Wed Aug 28 10:31:11 2002
@@ -1,5 +1,8 @@
 <?php
 
+while (list($k,$v)=each($_GET))$$k=$v;
+while (list($k,$v)=each($_POST))$$k=$v;
+
 /**
  * webmail.php -- Displays the main frameset
  *
