--- squirrelmail-1.2.8/plugins/username/setup.php.orig	Wed May  2 20:01:39 2001
+++ squirrelmail-1.2.8/plugins/username/setup.php	Fri Sep 20 14:21:37 2002
@@ -41,20 +41,27 @@
          return;
 	 
       ?>
-<table align=center cellpadding=0 cellspacing=0 border=0 bgcolor=<?php echo $color[10] ?>><tr><td>
-<table width=100% cellpadding=2 cellspacing=1 border=0 bgcolor="<?php echo $color[5] ?>"><tr><td align=center>
-<tt><?php
-      
+
+<table width=100% cellpadding=2 cellspacing=1 border=0 bgcolor="<?php echo $color[3] ?>"><tr><td align=center>
+<tt><h3><?php
+
       if ($show_username == 1)
-          echo $username;
-      if ($show_username == 2)
-          echo 'Logged in as: ' . $username;
-       
-?></tt>
-</td></tr></table>
+         echo $username;
+      if ($show_username == 2) {
+         $msgpl = 'Zalogowany jako: ' .  $username;
+	 $msgother = 'Logged in as: ' .  $username;
+	 if (getPref($data_dir, $username, 'language') == pl_PL)
+	    echo $msgpl;
+	 else if ($squirrelmail_default_language = 'pl_PL' && getPref($data_dir, $username, 'language') == '')
+	    echo $msgpl;
+	 else
+	    echo $msgother;
+	 }   
+
+?></h3></tt>
 </td></tr></table>
 <br>
-<?
+<?php
    }
 
    function username_show_options() {
@@ -105,10 +112,10 @@
    }
 
    function username_save_options() {
-      global $data_dir, $username, $username_in, $username_pos;
+      global $data_dir, $username, $username_in, $username_pos, $_POST;
 
-      setPref($data_dir, $username, 'show_username', $username_in);
-      setPref($data_dir, $username, 'show_username_pos', $username_pos);
+      setPref($data_dir, $username, 'show_username', $_POST['username_in']);
+      setPref($data_dir, $username, 'show_username_pos', $_POST['username_pos']);
    }
 
    function username_load_options() {
